### ISSUE TITLE
Mocks stop working when changing language due to path mismatch

### ISSUE THREAD

**Issue by indigane** [globaleaks/GlobaLeaks#3029]
**Describe the bug**
```js
GL.mockEngine.addMock('/#/', '#ContentBox', '<h1>Hello, world!</h1>', 'add-before');
```
This does not work when changing languages, because for example `?lang=en` is added to the end, making the `location.hash` `#/?lang=en`.

Which does not match, as `current_path` has to include the `location.hash` as here https://github.com/globaleaks/GlobaLeaks/blob/8b12592d04e47f9eee7f800764cef7d0b5adf4bc/client/app/js/mocks.js#L44

This issue appeared when updating from `v4.2.10` to `v4.2.11`.

**To Reproduce**
1. Set up a mock with a path **other than** `'*'`.
2. Change language from the UI.
3. `?lang=` is added to the hash.
4. Mock stops working because path no longer matches.

**Expected behavior**
Path based mocks should work without registering them separately for all the supported languages.

**Comment by evilaliv3** [comment]
Thank you for reporting this @indigane 

I think you are right and the issue could probably lie in the commit that you have identified. I will check this out and try to include a fix in the next release.

**Comment by evilaliv3** [comment]
@evilaliv3: I just retested your code alone ant it actually seems to work fine. Can you please retest it and provide me an alternative code that does actually fail?

**Comment by evilaliv3** [comment]
Nevermind @indigane i got to reproduce your issue using the full javascript of your site and i've identified how we could fix it. thank you

**Comment by evilaliv3** [comment]
The patch has been included in release 4.2.12
